PMM-2134: rds_exporter exports memory in Kb with node_exporter labels which are in bytes.

arvenil opened the pull request:
* Metrics multiplied by `1024`
  - node_disk_bytes_read
  - node_disk_bytes_written
  - node_filesystem_size
  - node_memory_Active
  - node_memory_Buffers
  - node_memory_Cached
  - node_memory_MemFree
  - node_memory_Inactive
  - node_memory_Mapped
  - node_memory_PageTables
  - node_memory_Slab
  - node_memory_MemTotal
  - node_memory_SwapFree
  - node_memory_SwapTotal
  - ~~node_vmstat_pswpin~~
  - ~~node_vmstat_pswpout~~
  - ~~node_vmstat_nr_dirty~~
* Renamed `node_memory_nr_dirty` to `node_vmstat_nr_dirty`

delgod commented on enhanced/metrics.go:
```diff
@@ -626,42 +694,47 @@ var Metrics = map[string]Metric{
 		),
+		Unit: 1024,
 	},
 	"node_vmstat_pswpin": {
```
hm, I am not sure, but maybe conversion is not needed here (for node_vmstat_pswpin and node_vmstat_pswpout).
please check

arvenil replied:
`node_vmstat_pswpin` and `node_vmstat_pswpout` are delivered from `swap->in` and `swap->out` which... don't have any description at https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Monitoring.OS.html but all `swap` metrics are described to be in kilobytes.

arvenil replied:
But we discussed it some time ago and we came with description that was saying it is in kilobytes:
https://percona.slack.com/archives/D3ZL51V0F/p1518785948000102

rnovikovP replied:
agree that, it'is very unlikely be something else  instead of kilobytes

delgod replied:
@rnovikovP @arvenil yes, we have kilobytes in AWS.
I think we have kilobytes also in node_exporter for these metrics...
it means - maybe conversion is not needed

adivinho replied:
node_exporter gives metrics node_vmstat_pswpin/out in pages.
So values are multiplied by page size (4096) in formulas.

rnovikovP replied:
oh, so it  should be converted to  pages in RDS exporter too, for now

arvenil replied:
@adivinho but then, we don't know in what unit is delivered metric from amazon for `node_vmstat_pswpin` and `node_vmstat_pswpout`. Those metrics are converted from `swap->in` and `swap->out` metrics from RDS, however they don't have description. On the other hand all metrics for `swap` are in kilobytes so maybe those are too.

delgod commented on enhanced/metrics.go:
```diff
@@ -365,51 +404,57 @@ var Metrics = map[string]Metric{
 		),
+		Unit: 1024,
 	},
 	"node_memory_nr_dirty": {
```
hm, I am not sure, but maybe conversion is not needed here.
please check

arvenil replied:
`node_memory_nr_dirty` is delivered from `memory->dirty` metric and it has description:
> The amount of memory pages in RAM that have been modified but not written to their related data block in storage, in kilobytes. 

(source: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Monitoring.OS.html)

delgod replied:
@rnovikovP @arvenil yes, we have kilobytes in AWS.
I think we have kilobytes also in node_exporter for these metrics...
it means - maybe conversion is not needed

adivinho replied:
node_exporter gives metrics node_vmstat_pswpin/out in pages.
So values are multiplied by page size (4096) in formulas.

adivinho replied:
node_memory_  metrics are collected in bytes by node_exporter

arvenil commented:
I'm not sure about any of those metrics, that's why I'm asking for second pair of eyes. Can we somehow verify those metrics empirically? Feature build?

arvenil commented:
@rnovikovP @delgod @adivinho just FYI https://github.com/prometheus/node_exporter/issues/895

arvenil commented:
@rnovikovP @delgod @adivinho I've reverted changes to questionable metrics so we  multiply by `1024` only below:
```
node_disk_bytes_read
node_disk_bytes_written
node_filesystem_size
node_memory_Active
node_memory_Buffers
node_memory_Cached
node_memory_MemFree
node_memory_Inactive
node_memory_Mapped
node_memory_PageTables
node_memory_Slab
node_memory_MemTotal
node_memory_SwapFree
node_memory_SwapTotal
```

Also renamed `node_memory_nr_dirty` to `node_vmstat_nr_dirty` as it was wrong name. Please re-review.

delgod commented on enhanced/generate/main.go:
```diff
@@ -54,6 +55,29 @@ func (m Metric) ConstLabels() map[string]string {
+	if strings.Contains(m.Help, "kilobytes") {
+		if strings.HasPrefix(m.FqName(), "node_") {
+			excluded := map[string]struct{}{
+				"in":    {},
```
is it enough to set `in` or it is needed to put `pswpin` here to exclude mertic convertion?

arvenil replied:
The metric here is called `swap_in`. It's just generator, ugly as hell at this point but does it job. Always verify the list of metrics, this is used on runtime.

arvenil commented:
@rnovikovP Should we merge that?

arvenil commented:
🎉
